core/lib/generators/spree/dummy/templates/rails/database.yml

```diff
@@ -33,17 +33,20 @@ production:
 development:
   adapter: postgresql
   database: <%= database_prefix %>spree_development
-  username: postgres
+  username: <%= ENV.fetch('DATABASE_USERNAME', 'default') %>
```

Comment:
to be backward compatible we should use `postgres` as the second argument. This way if no `DATABASE_USERNAME` is provided we fall back to that value.